core/libs/models/src/tree.rs

```diff
@@ -64,8 +66,16 @@ pub trait FileMetaExt<T: FileMetadata> {
     fn get_invalid_cycles(&self, staged_changes: &[T]) -> Result<Vec<Uuid>, TreeError>;
     fn get_path_conflicts(&self, staged_changes: &[T]) -> Result<Vec<PathConflict>, TreeError>;
     fn verify_integrity(&self) -> Result<(), TestFileTreeError>;
+    fn display(&self) -> Result<String, TreeError>;
 }
 
+// Why cant get work :(
+// impl Display for FileMetadata {
+//     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+//         self.display()
+//     }
+// }
+
```

Comment:
Alright good attempt here but what you want to do is something like this 
```
impl<Fm> Display for [Fm]
where
    Fm: FileMetadata,
```

I think something like this will work?

You want to express that you'll provide a static dispatch parameterized implementation based on what they asked for (think duplication in the binary).

I think if you impl'd that you could do away with the `display()` fn you authored entirely and you don't need your function to be a part of the `FileMetaExt` at all, it can just be a self standing sort of implementation. Slightly neater.